microservices/scraper-thaiNGO/scraper.py

```diff
@@ -0,0 +1,71 @@
+from bs4 import BeautifulSoup
+import requests
+from urllib.request import urlopen
+import csv
+
+baseurl = "http://wiki.p2pfoundation.net/NGOs_in_Thailand"
+
+"""
+list of dictionaries to store ngo information
+ngos = [{
+    name: "",
+    URL: "",
+    phone_num: "",
+    email: "",
+    registration_id: "",
+    year_established: ,
+    description: ""
+}]
+"""
+ngos = []
+ngos_store_keys = [
+    "ngo_name",
+    "ngo_URL",
+    "phone_num",
+    "email",
+    "registration_id",
+    "year_established",
+    "description",
+]
+
+"""
+DESCRIPTION: scrapes the base directory for links to specific Thai NGO wiki pages        
+"""
+
+
+def basepage_scrape():
+    page = requests.get(baseurl)
+    soup = BeautifulSoup(page.content, "html5lib")
+    # find all NGO entries represented as a row in html table
+    ngo_tr_tags = soup.find_all("tr")
+
+    for row in ngo_tr_tags:
+        # dictionary to store ngo information
+        ngo = {}
+        ngo_info = row.find_all("td")
+        if ngo_info is not None and len(ngo_info) > 0:
+            ngo_name = ngo_info[1]
+            ngo["name"] = ngo_name.text.lstrip()
+            ngo_descr = ngo_info[2]
+            ngo["description"] = ngo_descr.text.lstrip()
+            ngo["URL"] = ""
+            if ngo_info[3].a is not None:
+                ngo_URL = ngo_info[3].a.get("href")
+                ngo["URL"] = ngo_URL
+            # filter invalid email address formats
+            ngo["email"] = ""
+            if len(ngo_info[4].text) > 5 and "requested" not in ngo_info[4].text:
```

Comment:
Why does the length of the text in that field have to be greater than 5? Or what does 5 represent? I would try and remove hard coded numbers that aren't array indexes or 0/1 from the code and assign them to descriptive constants like MIN_EMAIL_LENGTH or something similar. This also allows for more modular/reusable code since changing the value in that const variable changes it in all places that you use it.